patternfly/patternfly-elements
# feat: primary details mobile optimization

## <Pull request description>

- Adds styles and behaviors necessary for a mobile optimized layout
  - Adds `breakpoint-width` attribute with a default value of 800, if the width of the component goes below this number it goes to mobile layout
  - Adds `breakpoint` attribute, displays what breakpoint the component is using "compact" or "desktop"
  - Added better focus styles
- Moved to new testing framework
  - Added more thorough testing
  - Got rid of files for old testing framework
  - Fixed errors found while writing tests
- Added docs
  - for new functionality
  - for a11y information
  - for new testing framework

### Related issues

- [CPFED-4084](https://projects.engineering.redhat.com/browse/CPFED-4084) Primary Detail looks bad at mobile


### Preview

Link(s) to demo page(s) where this element can be viewed:
- [Demo Page](https://deploy-preview-1582--patternfly-elements.netlify.app/elements/pfe-primary-detail/demo/) 


### Testing instructions



- [ ] *Check desktop layout*
  1. Open window at 1000px+ wide window
  2. Make sure layout is two column
  3. Click on nav items, make sure the content on the right changes
  4. Use keyboard navigation, when on tabs the arrow keys are used to navigate the tabs, everything else uses tabs
- [ ] *Check mobile layout*
  1. Open page at 500px or less wide window
  2. Make sure the first column (the nav side) is visible
  3. Click on nav item, it should slide over to the content side
  4. Click back or escape and it should go back to the nav side
- [ ] *Check going from one breakpoint to another*
  1. Make sure if something is open at desktop, it remains open at mobile
  2. Make sure that something is always open at desktop, the right column should never be empty (have nothing selected at mobile then go to desktop)
 
Run the tests on the component

Test in older browsers
 


#### Browser requirements

Your component should work in all of the following environments:



### Ready-for-merge Checklist





### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**
